api_tests/logs/views/test_log_detail.py

```diff
@@ -1,142 +1,200 @@
+        return url_logs + '{}/nodes/'.format(log_public._id)
 
-        self.user_one = AuthUserFactory()
-        self.user_two = AuthUserFactory()
+    @pytest.fixture()
+    def url_log_detail_private(self, log_private, url_logs):
+        return url_logs + '{}/'.format(log_private._id)
 
-        self.node = ProjectFactory(creator=self.user_one)
-        self.node.add_contributor(self.user_two)
+    @pytest.fixture()
+    def url_log_detail_public(self, log_public, url_logs):
+        return url_logs + '{}/'.format(log_public._id)
 
-        self.component = NodeFactory(parent=self.node, creator=self.user_one)
+@pytest.mark.django_db
+class TestLogDetail(LogsTestCase):
 
-        self.file = api_utils.create_test_file(node=self.component, user=self.user_one)
-        self.node.add_log(
+    def test_log_detail_private(self, app, url_log_detail_private, user_one, user_two, log_private):
+        #test_log_detail_returns_data
+        res = app.get(url_log_detail_private, auth=user_one.auth)
+        assert res.status_code == 200
+        json_data = res.json['data']
+        assert json_data['id'] == log_private._id
+
+        #test_log_detail_private_not_logged_in_cannot_access_logs
+        res = app.get(url_log_detail_private, expect_errors=True)
+        assert res.status_code == 401
+
+        #test_log_detail_private_non_contributor_cannot_access_logs
+        res = app.get(url_log_detail_private, auth=user_two.auth, expect_errors=True)
+        assert res.status_code == 403
+
+    def test_log_detail_public(self, app, url_log_detail_public, log_public, user_two, user_one):
+        #test_log_detail_public_not_logged_in_can_access_logs
+        res = app.get(url_log_detail_public, expect_errors=True)
+        assert res.status_code == 200
+        data = res.json['data']
+        assert data['id'] == log_public._id
+
+        #test_log_detail_public_non_contributor_can_access_logs
+        res = app.get(url_log_detail_public, auth=user_two.auth, expect_errors=True)
+        assert res.status_code == 200
+        data = res.json['data']
+        assert data['id'] == log_public._id
+
+        #test_log_detail_data_format_api
+        res = app.get(url_log_detail_public + '?format=api', auth=user_one.auth)
+        assert res.status_code == 200
+        assert log_public._id in unicode(res.body, 'utf-8')
+
+@pytest.mark.django_db
+class TestNodeFileLogDetail:
+
+    @pytest.fixture()
+    def user_one(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def user_two(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def node(self, user_one, user_two):
+        node = ProjectFactory(creator=user_one)
+        node.add_contributor(user_two)
+        return node
+
+    @pytest.fixture()
+    def component(self, user_one, node):
+        return NodeFactory(parent=node, creator=user_one)
+
+    @pytest.fixture()
+    def file(self, user_one, component):
```

Comment:
file is a type in python, probably won't cause errors, but should be renamed to something else just in case